### PR TITLE
[WIP] chore: update accelerated networking-capable SKU list

### LINKED ISSUE #1894: Automate accelerated networking VM sizes whitelist
https://github.com/Azure/aks-engine/blob/master/pkg/helpers/helpers.go#L118

At least Standard_D16_v3 is missing.

### PR DESCRIPTION
**Pull request by mboersma**
**Reason for Change**:
Updates the list of accelerated networking-capable VM SKUs. Specifically it adds these:

```
 'Standard_B12ms',
 'Standard_B16ms',
 'Standard_B20ms',
 'Standard_D16a_v3',
 'Standard_D16as_v3',
 'Standard_D32a_v3',
 'Standard_D32as_v3',
 'Standard_D48_v3',
 'Standard_D48a_v3',
 'Standard_D48as_v3',
 'Standard_D48s_v3',
 'Standard_D4a_v3',
 'Standard_D4as_v3',
 'Standard_D64a_v3',
 'Standard_D64as_v3',
 'Standard_D8a_v3',
 'Standard_D8as_v3',
 'Standard_DS11-1_v2',
 'Standard_DS12-1_v2',
 'Standard_DS12-2_v2',
 'Standard_DS12_v2',
 'Standard_DS13-2_v2',
 'Standard_DS3_v2',
 'Standard_E16-4s_v3',
 'Standard_E16-8s_v3',
 'Standard_E16a_v3',
 'Standard_E16as_v3',
 'Standard_E20_v3',
 'Standard_E20s_v3',
 'Standard_E32a_v3',
 'Standard_E32as_v3',
 'Standard_E4-2s_v3',
 'Standard_E48_v3',
 'Standard_E48a_v3',
 'Standard_E48as_v3',
 'Standard_E48s_v3',
 'Standard_E4a_v3',
 'Standard_E4as_v3',
 'Standard_E64a_v3',
 'Standard_E64as_v3',
 'Standard_E64i_v3',
 'Standard_E64is_v3',
 'Standard_E8-2s_v3',
 'Standard_E8-4s_v3',
 'Standard_E8a_v3',
 'Standard_E8as_v3',
 'Standard_F48s_v2',
 'Standard_L48s_v2',
 'Standard_L80s_v2',
 'Standard_M16-4ms',
 'Standard_M16-8ms',
 'Standard_M16ms',
 'Standard_M208ms_v2',
 'Standard_M208s_v2',
 'Standard_M32-16ms',
 'Standard_M32-8ms',
 'Standard_M32ms',
 'Standard_M416ms_v2',
 'Standard_M416s_v2',
 'Standard_M8-2ms',
 'Standard_M8-4ms',
 'Standard_M8ms'
```

Also updates the `generate_azure_constants.py` script to maintain this list in `azureconst.go`.

**Issue Fixed**:
Fixes #1894

**Requirements**:

- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Notes**:
TODO: sync up unit test coverage for gaps identified [here](https://github.com/Azure/aks-engine/pull/1913#issuecomment-528618860).

### CODE REVIEW

**Comment by jackfrancis** [comment]
A bunch of the SKUs in the original list aren't in the new one. Are we confident the new list is the accurate one?

File: pkg/helpers/generate_azure_constants.py [hunk 1]
```diff
@@ -75,7 +76,27 @@ def get_storage_account_type(size_name):
...
+def get_accelerated_skus():
+    """
+    Returns a dict of SKU name to AcceleratedNetworkingEnabled value.
+    """
+    pinned = [
+        ("AZAP_Performance_ComputeV17C", True),
+        ("SQLGL", True),
+        ("SQLGLCore", True),
+    ]
+    query = r"[? starts_with(name, `Standard`) && !ends_with(name, `Promo`)].{name: name, caps: capabilities[? name==`AcceleratedNetworkingEnabled`]}[].{sku: name, acceleratedNetworking: caps[0].value}[? acceleratedNetworking != null]"
```
Comment (jackfrancis) [inline pkg/helpers/generate_azure_constants.py]: Are we excluding promo SKUs to keep the list leaner?

Comment (mboersma) [reply pkg/helpers/generate_azure_constants.py]: Yes, basically. The `AcceleratedNetworkingSupported` func also starts with this line before interrogating the data structure:

```go
sku = strings.TrimSuffix(sku, "_Promo")
```

**Comment by mboersma** [comment]
> Are we confident the new list is the accurate one?

Apparently not! Several of the SKUs in the original list simply aren't returned by the `az vm list-skus` query. I can hard-code them in there as I did with "AZAP_Performance_ComputeV17C", but let me dig around for a bit first to see what's really going on.

**Comment by mboersma** [comment]
Those SKUs aren't listed as having any value for the `AcceleratedNetworkingEnabled` property, which I assumed is equivalent to `False`. I'll have to try them to see if they actually support accelerated networking, then add them to a grandfathered list if so.

**Comment by mboersma** [comment]
After updating the strategies to locate accelerated-networking-capable SKUs, these are the additions:
```python
In [9]: set(new) - set(old)                                                                                                       
Out[9]: 
{'Standard_B12ms',
 'Standard_B16ms',
 'Standard_B20ms',
 'Standard_D16a_v3',
 'Standard_D16as_v3',
 'Standard_D32a_v3',
 'Standard_D32as_v3',
 'Standard_D48_v3',
 'Standard_D48a_v3',
 'Standard_D48as_v3',
 'Standard_D48s_v3',
 'Standard_D4a_v3',
 'Standard_D4as_v3',
 'Standard_D64a_v3',
 'Standard_D64as_v3',
 'Standard_D8a_v3',
 'Standard_D8as_v3',
 'Standard_DS11-1_v2',
 'Standard_DS12-1_v2',
 'Standard_DS12-2_v2',
 'Standard_DS12_v2',
 'Standard_DS13-2_v2',
 'Standard_DS3_v2',
 'Standard_E16-4s_v3',
 'Standard_E16-8s_v3',
 'Standard_E16a_v3',
 'Standard_E16as_v3',
 'Standard_E20_v3',
 'Standard_E20s_v3',
 'Standard_E32a_v3',
 'Standard_E32as_v3',
 'Standard_E4-2s_v3',
 'Standard_E48_v3',
 'Standard_E48a_v3',
 'Standard_E48as_v3',
 'Standard_E48s_v3',
 'Standard_E4a_v3',
 'Standard_E4as_v3',
 'Standard_E64a_v3',
 'Standard_E64as_v3',
 'Standard_E64i_v3',
 'Standard_E64is_v3',
 'Standard_E8-2s_v3',
 'Standard_E8-4s_v3',
 'Standard_E8a_v3',
 'Standard_E8as_v3',
 'Standard_F2s_v2',
 'Standard_F48s_v2',
 'Standard_L48s_v2',
 'Standard_L80s_v2',
 'Standard_M128',
 'Standard_M128m',
 'Standard_M16-4ms',
 'Standard_M16-8ms',
 'Standard_M16ms',
 'Standard_M208ms_v2',
 'Standard_M208s_v2',
 'Standard_M32-16ms',
 'Standard_M32-8ms',
 'Standard_M32ls',
 'Standard_M32ms',
 'Standard_M32ts',
 'Standard_M416ms_v2',
 'Standard_M416s_v2',
 'Standard_M64',
 'Standard_M64ls',
 'Standard_M64m',
 'Standard_M8-2ms',
 'Standard_M8-4ms',
 'Standard_M8ms'}
```
and these are removed:
```python
In [10]: set(old) - set(new)                                                                                                      
Out[10]: 
{'Standard_D12_v2_ABC',
 'Standard_D13_v2_ABC',
 'Standard_D14_v2_ABC',
 'Standard_D15_v2_ABC',
 'Standard_D32-16s_v3',
 'Standard_D32-8s_v3',
 'Standard_D3_v2_ABC',
 'Standard_D40_v3',
 'Standard_D40s_v3',
 'Standard_D4_v2_ABC',
 'Standard_D5_v2_ABC',
 'Standard_D64-16s_v3',
 'Standard_D64-32s_v3',
 'Standard_E32-16_v3',
 'Standard_F16_ABC',
 'Standard_F4_ABC',
 'Standard_F8_ABC',
 'Standard_L96s_v2'}
```

I can't see any of the latter set of SKUs in the portal anywhere, nor through `az vm list-skus --all`. The names with an `_ABC` suffix don't sound like production names to me. Maybe these went away, or maybe they're only available to a subscription in special cases?

It's probably safest to add them to the initial list of grandfathered SKUs, although if anyone knows that they can be omitted, please let me know.

**Comment by mboersma** [comment]
Ok, I found a couple regex bugs and some outliers, and updated the grandfathered list so no SKUs are removed as part of this PR, only added:

```python
In [6]: set(new) - set(old)                                                                                                       
Out[6]: 
{'Standard_B12ms',
 'Standard_B16ms',
 'Standard_B20ms',
 'Standard_D16a_v3',
 'Standard_D16as_v3',
 'Standard_D32a_v3',
 'Standard_D32as_v3',
 'Standard_D48_v3',
 'Standard_D48a_v3',
 'Standard_D48as_v3',
 'Standard_D48s_v3',
 'Standard_D4a_v3',
 'Standard_D4as_v3',
 'Standard_D64a_v3',
 'Standard_D64as_v3',
 'Standard_D8a_v3',
 'Standard_D8as_v3',
 'Standard_DS11-1_v2',
 'Standard_DS12-1_v2',
 'Standard_DS12-2_v2',
 'Standard_DS12_v2',
 'Standard_DS13-2_v2',
 'Standard_DS3_v2',
 'Standard_E16-4s_v3',
 'Standard_E16-8s_v3',
 'Standard_E16a_v3',
 'Standard_E16as_v3',
 'Standard_E20_v3',
 'Standard_E20s_v3',
 'Standard_E32a_v3',
 'Standard_E32as_v3',
 'Standard_E4-2s_v3',
 'Standard_E48_v3',
 'Standard_E48a_v3',
 'Standard_E48as_v3',
 'Standard_E48s_v3',
 'Standard_E4a_v3',
 'Standard_E4as_v3',
 'Standard_E64a_v3',
 'Standard_E64as_v3',
 'Standard_E64i_v3',
 'Standard_E64is_v3',
 'Standard_E8-2s_v3',
 'Standard_E8-4s_v3',
 'Standard_E8a_v3',
 'Standard_E8as_v3',
 'Standard_F48s_v2',
 'Standard_L48s_v2',
 'Standard_L80s_v2',
 'Standard_M16-4ms',
 'Standard_M16-8ms',
 'Standard_M16ms',
 'Standard_M208ms_v2',
 'Standard_M208s_v2',
 'Standard_M32-16ms',
 'Standard_M32-8ms',
 'Standard_M32ms',
 'Standard_M416ms_v2',
 'Standard_M416s_v2',
 'Standard_M8-2ms',
 'Standard_M8-4ms',
 'Standard_M8ms'}
```

**Pull request closed without merging**